models/user/user.go

```diff
@@ -729,6 +734,15 @@ func createUser(ctx context.Context, u *User, createdByAdmin bool, overwriteDefa
 	return committer.Commit()
 }
 
+func HitCreationLimit(ctx context.Context) bool {
+	// don't bother calling DB if limit not set
+	if setting.Service.MaxUserCreationLimit == -1 ||
```

Comment:
I've wondered whether we should allow any negative value as `unrestricted`, and concluded that I'm fine with both approaches.

Reply:
Ah wait, there is currently a bug for anything negative apart from `-1`: It means that no users can be created at all, so it behaves like `0`.
That is unintended, so we need to check for `>= 0`

Reply:
```suggestion
	if setting.Service.MaxUserCreationLimit < 0 ||
```